pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -412,13 +411,20 @@ public UnAckedMessageTracker getUnAckedMessageTracker() {
         try {
             message = incomingMessages.take();
             messageProcessed(message);
+            if (checkMessageImplConsumerEpochIsSmallerThanConsumer(message)) {
```

Comment:
```suggestion
            if (!isValidConsumerEpoch(message)) {
```

More straightforward and understandable